File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandMoveToLastBlock.java
package com.kota.Bahamut.Command;

import androidx.annotation.NonNull;

import com.kota.Bahamut.ListPage.TelnetListPage;
import com.kota.Bahamut.ListPage.TelnetListPageBlock;

public class BahamutCommandMoveToLastBlock extends BahamutCommandLoadLastBlock {
    public BahamutCommandMoveToLastBlock() {
        this.Action = MoveToLastBlock;
    }

    public void executeFinished(TelnetListPage aListPage, TelnetListPageBlock aPageData) {
        super.executeFinished(aListPage, aPageData);
        aListPage.pushRefreshCommand(1);
        setDone(true);
    }

    @NonNull
    public String toString() {
        return "[MoveToLastBlock]";
    }
}

File: app/src/main/java/com/kota/Bahamut/command/BahamutCommandMoveToLastBlock.kt
package com.kota.Bahamut.command

import com.kota.Bahamut.listPage.TelnetListPage
import com.kota.Bahamut.listPage.TelnetListPageBlock

class BahamutCommandMoveToLastBlock : BahamutCommandLoadLastBlock() {
    init {
        this.action = BahamutCommandDef.Companion.MOVE_TO_LAST_BLOCK
    }

    override fun executeFinished(telnetListPage: TelnetListPage, telnetListPageBlock: TelnetListPageBlock?) {
        super.executeFinished(telnetListPage, telnetListPageBlock)
        telnetListPage.pushRefreshCommand(1)
        isDone = true
    }

    override fun toString(): String {
        return "[MoveToLastBlock]"
    }
}
